### PR TITLE
LPS-89170

### PR DESCRIPTION
**Pull request by danielkocsis**
Hey Preston,

finally I was able to spend some time with it. Please let me know your opinion and suggestions if you think I need to change something!

thanks,
Daniel

### CODE REVIEW

**Comment by danielkocsis** [comment]
ci:test:relevant

**Comment by danielkocsis** [comment]
ci:test:sf

**Comment by danielkocsis** [comment]
Hey @Preston-Crary,

it seems the upgrade issue is mine, so i'll check it and resend, but feel free to check the other changes, and please let me know your findings, so i'll fix those as well.

thanks,
Daniel

File: sql/indexes.sql [hunk 1]
```diff
@@ -225,6 +225,7 @@ create index IX_8EC3D2BC on LayoutRevision (plid, status);
 create index IX_421223B1 on LayoutRevision (status);
 
 create unique index IX_48550691 on LayoutSet (groupId, privateLayout);
```
Comment (Preston-Crary) [inline sql/indexes.sql]: Need to manually remove the unique indexes from LayoutSet to have them be updated during regenerate.  This should be "create unique index IX_*** on LayoutSet (groupId, privateLayout, head);"

File: sql/indexes.sql [hunk 2]
```diff
@@ -224,7 +224,7 @@ create index IX_7FFAE700 on LayoutRevision (layoutSetBranchId, status);
 create index IX_8EC3D2BC on LayoutRevision (plid, status);
 create index IX_421223B1 on LayoutRevision (status);
 
-create unique index IX_48550691 on LayoutSet (groupId, privateLayout);
```
Comment (Preston-Crary) [inline sql/indexes.sql]: Nevermind, I see the change here.  Can you merge the two generated commits?

**Comment by Preston-Crary** [comment]
Can we also merge e883c29 and 230bf7a so it's clearer what changes were done to support versioning?

**Comment by Preston-Crary** [comment]
I think we have a problem whenever using a unique delete method like here https://github.com/Preston-Crary/liferay-portal/blob/852a5b283fd07b51a88622b79eb245f630f56cf3/portal-impl/src/com/liferay/portal/service/impl/LayoutSetLocalServiceImpl.java#L133

I should add some logic to the ServiceBuilder to avoid generating these persistence methods, we need to call delete(layoutSet) otherwise we can end up orphaning the draft and versions.

File: portal-impl/src/com/liferay/portal/service/impl/LayoutSetLocalServiceImpl.java [hunk 3]
```diff
@@ -397,12 +377,14 @@ public LayoutSet updateLookAndFeel(
...
-			layoutSet.setCss(css);
+			LayoutSet draftLayoutSet = getDraft(layoutSet);
+
+			draftLayoutSet.setModifiedDate(new Date());
+			draftLayoutSet.setThemeId(themeId);
+			draftLayoutSet.setColorSchemeId(colorSchemeId);
+			draftLayoutSet.setCss(css);
 
-			layoutSetPersistence.update(layoutSet);
+			publishDraft(draftLayoutSet);
```
Comment (Preston-Crary) [inline portal-impl/src/com/liferay/portal/service/impl/LayoutSetLocalServiceImpl.java]: Let's do this here and then return layoutSet like before:
layoutSet = publishDraft(draftLayoutSet);

**Comment by Preston-Crary** [comment]
Need to do the getDraft pattern here as well:
https://github.com/Preston-Crary/liferay-portal/blob/852a5b283fd07b51a88622b79eb245f630f56cf3/portal-impl/src/com/liferay/portal/service/impl/LayoutSetLocalServiceImpl.java#L438

**Comment by Preston-Crary** [comment]
Hi @danielkocsis,

I took a look, please see my comments and update the code after you resolve the upgrade issues.

Thanks!

**Comment by Preston-Crary** [comment]
I'll try to make time to remove the generated persistence deletion methods when using versioning, but you can get around that by just not calling them anymore.

**Pull request closed without merging**